### PR TITLE
firefox: add support for specifying policies

### PR DESCRIPTION
**Pull request by DamienCassou**
### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

#### Maintainer CC

/cc @rycee @kira-bruneau

### CODE REVIEW

**Comment by Nyabinary** [comment]
Why is this marked as draft, what needs to be added?

**Comment by DamienCassou** [comment]
> Why is this marked as draft, what needs to be added?

Thank you for noticing. This PR was waiting for https://github.com/nix-community/home-manager/pull/4633 to be merged. Now that it has, I added a missing test file and the PR is now ready to be reviewed.

**Comment by Nyabinary** [comment]
@rycee Seems good to merge right?

File: modules/programs/firefox.nix [hunk 1]
```diff
@@ -169,7 +169,10 @@ let
     else if isDarwin then
       package
     else if versionAtLeast config.home.stateVersion "19.09" then
-      package.override (old: { cfg = old.cfg or { } // fcfg; })
+      package.override (old: {
+        cfg = old.cfg or { } // fcfg;
+        extraPolicies = cfg.policies;
```
Comment (jian-lin) [inline modules/programs/firefox.nix:174]: This is a breaking change if users use `programs.firefox.package = pkgs.firefox.override { extraPolicies = { ... } }` to specify their policies before.  A release note should be added.

Comment (DamienCassou) [reply modules/programs/firefox.nix:174]: @rycee this change is part of the already released 23.11. Should old release notes be modified?

Comment (jian-lin) [reply modules/programs/firefox.nix:174]: Changing the implementation to `extraPolicies = old.extraPolicies // cfg.policies;` makes this almost not breaking.

Comment (rycee) [reply modules/programs/firefox.nix:174]: Adding to the existing extra policies seems like a good idea. Perhaps need to add a guard like in the `cfg` line above, though? I.e.

``` nix
extraPolicies = old.extraPolicies or { } // cfg.policies;
```

Does that seem ok? With that change I guess it wouldn't be necessary to add to the changelog? Regardless, I have nothing against adding to the 23.11 changelog, if it helps clarify…

**Pull request merged**